scripts/adb_devices_abis.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+set -e
+typeset -a abis
+SCRIPT_NAME=$(basename $0)
+
+function debug() {
+  echo "${SCRIPT_NAME} DEBUG: ${@}" >&2
+}
+
+# make sure server is running, otherwise sdb will show devices offline
+adb start-server
+
+while IFS= read line; do
+  read -a arr <<< "${line}"
+
+  device_name="${arr[0]}"
+  device_status=${arr[1]}
+
+  debug "device_name=${device_name}"
+  debug "device_status=${device_status}"
+
+  if [[ "${device_status}" != "device" ]]; then
+    continue
+  fi
+
+  device_product_abi=$(adb -s ${device_name} shell -n getprop ro.product.cpu.abi)
+  debug "device_product_abi=${device_product_abi}"
+
+  if [[ ! ${abis[*]} =~ ${device_product_abi} ]]; then
+    abis+=("${device_product_abi}")
+  fi
+done <<< "$(adb devices | tail -n+2)"
+
+if [[ ! "${abis[*]}" ]]; then
+  echo "${SCRIPT_NAME}: ERROR: no devices found. Check 'adb devices -l' output and share with Infra team if needed." >&2
+  exit 3
+fi
+
+echo "${SCRIPT_NAME} DEBUG: resulting abis:" >&2
```

Comment:
Not using `debug` function. Also, we could just have a `log` function that takes 2 arguments, level and message.